pyproject.toml

```diff
@@ -0,0 +1,103 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Testing tools configuration
+[tool.coverage.run]
+branch = true
+
+[tool.coverage.report]
+show_missing = true
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+log_cli_level = "INFO"
+asyncio_mode = "auto"
+markers = ["unstable"]
+
+# Formatting tools configuration
+[tool.black]
+line-length = 99
+target-version = ["py310"]
+
+[tool.isort]
+profile = "black"
+
+[tool.poetry]
+name = "spark-configuration-hub"
+version = "1.0"
+description = "spark-configuration-hub"
+authors = []
+
+[tool.poetry.dependencies]
+python = ">=3.8,<4.0"
+ops = ">=2.4.1"
+pydantic ="^1.10.7"
```

Comment:
are we using pydantic anywhere? I may have missed this, but it seems to me that it is used only on integration tests (and there we could use dataclasses, although I really like pydantic...)